analysers/analyser_osmosis_polygon_intersects.py

```diff
@@ -98,37 +97,38 @@
   ST_Area(ST_MakePolygon(ST_Transform(linestring, {proj}))) >= 20
 UNION ALL
 SELECT
-  ways.id,
-  ways.linestring,
-  -- Better would be to parse inner and outer segments and create an SQL multipolygon
-  -- For now, just use a very small buffer zone around the linestring
-  ST_Buffer(ways.linestring, 0.0000000001, 'quad_segs=1 endcap=flat join=bevel') AS poly,
+  'R' || mp.id AS id,
+  ST_Transform(ST_LineMerge(ST_Collect(ways.linestring)), {proj}) AS linestring_proj, -- usually multilinestring
+  mp.poly,
   CASE
-    WHEN r.tags?'landuse' THEN 'landuse'
-    WHEN r.tags?'natural' THEN 'natural'
+    WHEN mp.tags?'landuse' THEN 'landuse'
+    WHEN mp.tags?'natural' THEN 'natural'
   END AS landusekey,
   CASE
-    WHEN r.tags?'landuse' THEN r.tags->'landuse'
-    WHEN r.tags?'natural' THEN r.tags->'natural'
+    WHEN mp.tags?'landuse' THEN mp.tags->'landuse'
+    WHEN mp.tags?'natural' THEN mp.tags->'natural'
   END AS landusevalue,
-  r.tags,
-  TRUE AS is_multipolygon
+  mp.tags
 FROM
-  relation_members
-  LEFT JOIN relations AS r ON
-    relation_id = r.id
+  multipolygons AS mp
+  JOIN relation_members ON
+    relation_id = mp.id
```

Comment:
No sure why we still need relation members?

Reply:
This particular analyser filters out minor intersections too. I do not have a good way to find (and filter) all minor intersections based on the geometry only.
I'm not referring to perpendicular incursions (such as the one on the left in the example picture) as they can easily be filtered by length, but are relatively rare. Instead I'm referring to sloped, shallow, incursions, such as the one on top (which constitutes the majority of the small incursions). They can have a very long length, yet only penetrate half a meter at most. This latter type is filtered based on the proximity to the borders of the relation instead. And to get the borders, I need the linestrings. See line 182 and 183 in the code

![image](https://github.com/osm-fr/osmose-backend/assets/1315520/cdb0d88c-6e74-47a2-9b95-bb22c9f9b9b8)

Reply:
Ok, but now we have MP as valid polygon, we can just use a negative ST_Buffer to adress both cases, cannot ?

Reply:
Good point, hadn't thought of that method.